utils/compression/zstd_compressor.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package compression
+
+import (
+	"github.com/DataDog/zstd"
+)
+
+var _ Compressor = (*zstdCompressor)(nil)
+
+func NewZstdCompressor() Compressor {
+	return &zstdCompressor{}
+}
+
+type zstdCompressor struct{}
+
+func (*zstdCompressor) Compress(msg []byte) ([]byte, error) {
+	return zstd.Compress(nil, msg)
```

Comment:
We should probably handle the max message size restrictions here like we did with gzip

Reply:
added